src/execution/command_parser.c

```diff
@@ -6,82 +6,156 @@
 #include "DAG/dag_parser.h"
 #include "util/string_utils.h"
 #include "execution/modelRun_ctx.h"
+#include "deprecated.h"
+
+extern int rlecMajorVersion;
+
+static inline int IsEnterprise() { return rlecMajorVersion != -1; }
+
+// Use this to check if a command is given a key whose hash slot is not on the current
+// shard, when using enterprise cluster.
+static void _AnalyzeKey(RedisModuleString *key_str, RAI_Error *err) {
```

Comment:
move to utils
change to something like `bool checkKeyInShardSlot(RedisModuleString *key)`
set the error outside this function